Applications/yOCTScanAndFindTissueSurface.m

```diff
@@ -1,44 +1,143 @@
+
+%% Reconstruct OCT Image for Subsequent Surface Analysis
+fprintf('Generating image...\n');
+outputTiffFile = [output_folder '\surface_analysis.tiff'];
+yOCTProcessTiledScan(...
+    volumeOutputFolder, ... Input
+    {outputTiffFile},... Save only Tiff file as folder will be generated after smoothing
+    'focusPositionInImageZpix',focusPositionInImageZpix,...
+    'dispersionQuadraticTerm',dispersionQuadraticTerm,...
+    'cropZAroundFocusArea', false,...
+    'v',v);
+[logMeanAbs, dimensions, ~] = yOCTFromTif(outputTiffFile);
+if (v)
+    disp([newline, '-- Data loaded successfully.']);
+end
 
 %% Estimate tissue surface
-% Todo
+if (v)
+    fprintf('%s Identifying tissue surface...\n', datestr(datetime));
+end
+tic;
+dimensions = yOCTChangeDimensionsStructureUnits(dimensions,'millimeters'); % Make sure it's in mm
+surfacePosition_mm = yOCTFindTissueSurface(logMeanAbs, dimensions, 'isVisualize', isVisualize);
+x_mm=dimensions.x.values;
+y_mm=dimensions.y.values;
+elapsedTimeSurfaceDetection = toc;
+if (v)
+    fprintf('%s Surface identification completed in %.2f seconds.\n', datestr(datetime), elapsedTimeSurfaceDetection);
+end
 
 %% Clean up
-% delete temporary files
+if exist(output_folder, 'dir')
+    rmdir(output_folder, 's'); % Remove the output directory after processing
+end
+totalEndTime = datetime;  % Capture the ending time
+totalDuration = totalEndTime - totalStartTime;
+if (v)
+    fprintf('Total surface analysis process completed in %s.\n', datestr(totalDuration, 'HH:MM:SS'));
+end
 
-%% Return
-% Tbd, this is just a demo
-surfacePosition_mm = ones(10,10);
-x_mm=1:10;
-y_mm=1:10;
+%% Check if the average surface distance is within the acceptable range
+average_surface_distance = nanmean(surfacePosition_mm(:)); % Calculate the average surface distance
+if isnan(average_surface_distance)
+    warning(['No tissue identification possible. Likely the tissue is out of focus below the detection range. ' ...
+        'Please manually increase the stage Z position to bring the tissue into focus.']);
+    isSurfaceInFocus = false; % Unable to verify if tissue surface is at focus
+elseif abs(average_surface_distance) > acceptableRange_mm
+    if average_surface_distance > 0 % Determine direction of adjustment
+        direction = 'increase';
+    else
+        direction = 'decrease';
+    end
+    warning('off', 'backtrace'); % Temporarily turn off backtrace for clear message formatting
+    warning(sprintf(['The average distance of the surface (%.4f mm) is out of range.\n\n', ...
+    'Please %s the stage Z position by %.2f mm to bring the tissue surface into focus. \n'], ...
+    average_surface_distance, direction, abs(round(average_surface_distance, 2))));
+    warning('on', 'backtrace'); % Turn backtrace back on
+    isSurfaceInFocus = false; % Tissue surface is not at focus
+else
+    if (v)
+        disp(['The average distance of the surface (', num2str(average_surface_distance, '%.4f'), ' mm) is within the acceptable range.']);
```

Comment:
Format - use this:
fprintf('%s WHATEVER YOU WANT TO SAY.\n', datestr(datetime))